File: RepoTransBench/target_projects/Java/C#/sockeqwe_fragmentargs/public_tests/ArgsBundlerInterfacePublicTest.cs
using Xunit;

namespace ProjectName.PublicTests.Bundler
{
    public class Bundle
    {
        private readonly Dictionary<string, string> _stringDict = new();

        public void PutString(string key, string value) => _stringDict[key] = value;
        public string? GetString(string key) => _stringDict.TryGetValue(key, out var v) ? v : null;
    }

    public abstract class ArgsBundler<T>
    {
        public abstract void Put(string key, T value, Bundle bundle);
        public abstract T Get(string key, Bundle bundle);
    }

    public class PublicTestBundler : ArgsBundler<string>
    {
        public override void Put(string key, string value, Bundle bundle)
        {
            bundle.PutString(key, "PUBLIC_" + value + "_PUBTEST");
        }

        public override string Get(string key, Bundle bundle)
        {
            var v = bundle.GetString(key);
            return v == null ? null! : v.Replace("PUBLIC_", "").Replace("_PUBTEST", "");
        }
    }

    public class ArgsBundlerInterfacePublicTest
    {
        [Fact]
        public void InterfacePutAddsPrefixSuffix_Public()
        {
            var bundle = new Bundle();
            new PublicTestBundler().Put("PUBKEY", "valueForPublic", bundle);
            Assert.Equal("PUBLIC_valueForPublic_PUBTEST", bundle.GetString("PUBKEY"));
        }

        [Fact]
        public void InterfaceGetRemovesPrefixSuffix_Public()
        {
            var bundle = new Bundle();
            bundle.PutString("PUB_KEY", "PUBLIC_zxy_PUBTEST");
            Assert.Equal("zxy", new PublicTestBundler().Get("PUB_KEY", bundle));
        }
    }
}

File: RepoTransBench/target_projects/Java/Go/sockeqwe_fragmentargs/public_tests/argsbundlerinterface_public_test.go
package public_tests

import (
	"testing"
)

type BundlePublic map[string]interface{}

type PublicTestBundler struct{}

func (b *PublicTestBundler) Put(key string, value string, bundle BundlePublic) {
	bundle[key] = "PUBLIC_" + value + "_PUBTEST"
}

func (b *PublicTestBundler) Get(key string, bundle BundlePublic) string {
	v, ok := bundle[key].(string)
	if !ok {
		return ""
	}
	return stripPrefixSuffix(v)
}

func stripPrefixSuffix(val string) string {
	if len(val) >= 14 && val[:7] == "PUBLIC_" && val[len(val)-8:] == "_PUBTEST" {
		return val[7 : len(val)-8]
	}
	return val
}

func TestInterfacePutAddsPrefixSuffixPublic(t *testing.T) {
	bundle := BundlePublic{}
	(&PublicTestBundler{}).Put("PUBKEY", "valueForPublic", bundle)
	expected := "PUBLIC_valueForPublic_PUBTEST"
	if bundle["PUBKEY"] != expected {
		t.Errorf("Expected %q, got %q", expected, bundle["PUBKEY"])
	}
}

func TestInterfaceGetRemovesPrefixSuffixPublic(t *testing.T) {
	bundle := BundlePublic{"PUB_KEY": "PUBLIC_zxy_PUBTEST"}
	got := (&PublicTestBundler{}).Get("PUB_KEY", bundle)
	if got != "zxy" {
		t.Errorf("Expected \"zxy\", got %q", got)
	}
}